opte/src/engine/dhcp.rs

```diff
@@ -243,6 +244,35 @@ impl ClasslessStaticRouteOpt {
     }
 }
 
+const DOMAIN_SEARCH_OPTION_OPTION_CODE: u8 = 119;
+const MAX_OPTION_LEN: usize = 255;
+
+// Encoded a list of Domain Names into a Domain Search Option, in accordance
+// with RFC 3397.
+//
+// Note that this may return more than one actual DHCP option. Those are
+// limited to 255 octets. However, domain names can be long, and we don't
+// want to artificially constrain the number of names that can be sent in a
+// DHCP Lease. Thus we allow any number, and encode them into as many
+// underlying options as required.
+fn encode_domain_search_option(names: &[DomainName]) -> Vec<u8> {
+    let mut all_names = Vec::new();
+    for name in names {
+        all_names.extend_from_slice(name.encode());
+    }
+
+    let mut out = Vec::with_capacity(all_names.len());
```

Comment:
That's correct. The issue is that `all_names` could be longer than the 255-maximum for a single DHCP option. This is pretty close, but you're right, it's not exact. I'm hoping that reallocations are rare or non-existent, e.g., since reservation and resizing are amortized O(1).

But, we _could_ make this exact by computing the number of 255-byte chunks, and then adding space for the option code for each. I'll experiment now, see if that is easy enough. Thanks!

Reply:
I think I've got a reasonably clear way to compute the correct size, will be in the next commit. Thanks!

Reply:
Ok, the version in [83206c5](https://github.com/oxidecomputer/opte/pull/324/commits/83206c5812301152d943a3ce1cdfb8fca71f645b) is not correct. I'm still experimenting.

Reply:
Think I got it right in `1e35fe0`